entity/dns_resource.go

```diff
@@ -16,3 +16,10 @@ type DNSResourceParams struct {
 	IPAddresses string `url:"ip_addresses,omitempty"`
 	AddressTTL  int    `url:"address_ttl"`
 }
+
+type DNSResourcesParams struct {
+	All    bool   `url:"all,omitempty"`
```

Comment:
https://github.com/canonical/maas/blob/master/src/maasserver/api/dnsresources.py#L155 based on MAAS code, `fqdn` is also a valid parameter. Could you please add it? In addition, if you are up for a MAAS contribution, you can try to open an MP to include it at the pydoc above. It will then be reflected to our API documentation.